framework/Optimizers/Optimizer.py

```diff
@@ -908,7 +907,7 @@ def generateInput(self,model,oldInput):
     self.localGenerateInput(model,oldInput)
     ####   UPDATE STATICS   ####
     # get trajectory asking for eval from LGI variable set
-    traj = self.inputInfo['trajectory']
+    traj = self.inputInfo['trajID'] - 1
```

Comment:
Sorry, I did not get the point here?

Reply:
Historically, the "trajID" was stored as "trajectory + 1" for limitations in the old HistorySet.  It's easier to keep that tradition than try to change it everywhere in the optimizer.